hack/v1migration/phase1.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+
+set -e
+
+if [ $# != 1 ]; then
+	echo "USAGE: $0 [WORKDIR]"
+	exit 1
+fi
+
+WORKDIR=$1
+echo "Using workdir: $WORKDIR"
+
+oc project hive
+
+oc scale -n hive deployment.v1.apps/hive-operator --replicas=0
+oc scale -n hive deployment.v1.apps/hive-controllers --replicas=0
+oc scale -n hive deployment.v1.apps/hiveadmission --replicas=0
+
+# TODO: wait until no pods are running
+
+mkdir -p $WORKDIR/
+
+HIVE_TYPES=( checkpoints clusterdeployments clusterdeprovisionrequests clusterimagesets clusterprovisions clusterstates dnsendpoints dnszones hiveconfig selectorsyncidentityprovider selectorsyncset syncidentityprovider syncsetinstance syncset )
+for i in "${HIVE_TYPES[@]}"
+do
+	:
+	echo "Storing all ${i} in ${WORKDIR}/${i}.yaml"
```

Comment:
How would you feel about storing each resource in its own file rather than one file per kind?

Reply:
I'd like to keep the scripts as simple as possible, is this easy to do? I can't immediately think of a way aside from querying one at a time, the I'm not crazy about the jq solutions to split I'm seeing.

Are there specific benefits? I couldn't think of any.

Reply:
My two motivations are (1) making it easier to manually validate the data collected and (2) making it easier to re-apply the resources back.